### PR TITLE
support retain chunkfilepool when clean

### PR DESCRIPTION
**Pull request by bai-charisu**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curve-ansible/roles/clean/tasks/include/clean_chunkserver.yml [hunk 1]
```diff
@@ -22,3 +22,7 @@
 - name: clean chunkserver without format disk
   include_tasks: "include/clean_chunkserver_without_disk_format.yml"
   when: not {{ chunkserver_format_disk }}
+
+- name: clean log file
```
Comment (ilixiaocui) [inline curve-ansible/roles/clean/tasks/include/clean_chunkserver.yml:26]: clean chunkserver with format disk

会把log删掉的，为啥不直接放到任务的yml中

File: curve-ansible/roles/clean/files/recycle_chunks.sh [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        do
+            maxChunkId=$(($maxChunkId+1))
+            sudo mv ${copyset}/data/${chunk} ../chunkfilepool/${maxChunkId}
+        done
+        # recycle wal segments
+        segments=`ls ${copyset}/log | grep curve_log`
+        for segment in $segments
+        do
+            maxSegmentId=$(($maxSegmentId+1))
+            sudo mv ${copyset}/log/${segment} ../chunkfilepool/${maxSegmentId}
```
Comment (ilixiaocui) [inline curve-ansible/roles/clean/files/recycle_chunks.sh:58]: sudo mv ${copyset}/log/${segment} ../walfilepool/${maxSegmentId} 吧

File: curve-ansible/roles/clean/tasks/include/clean_chunkserver_totally.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  with_sequence: start=0 end={{ end_seq }}
+
+- name: umount dir
+  ignore_errors: true
+  vars:
+    end_seq: "{{ chunkserver_num | int - 1}}"
+  shell: sudo umount {{ chunkserver_data_dir }}/chunkserver{{ item }}
+  with_sequence: start=0 end={{ end_seq }}
+
+- name: clear data dir
```
Comment (ilixiaocui) [inline curve-ansible/roles/clean/tasks/include/clean_chunkserver_totally.yml:31]: 重复了

File: curve-ansible/roles/clean/tasks/include/clean_chunkserver_with_disk_format.yml [hunk 4]
```diff
@@ -15,21 +15,10 @@
...
 #
 
-- name: clear data dir contents and log dir of each chunkserver
-  vars:
-    end_seq: "{{ chunkserver_num | int - 1}}"
-  shell: sudo rm -rf {{ chunkserver_data_dir }}/chunkserver{{ item }}/* && sudo rm -rf {{ chunkserver_data_dir }}/log/chunkserver{{ item }}
-  with_sequence: start=0 end={{ end_seq }}
+- name: clean chunkserver but retain chunkfilepool
+  include_tasks: clean_chunkserver_retain_chunkfilepool.yml
+  when: retain_chunkfilepool
```
Comment (ilixiaocui) [inline curve-ansible/roles/clean/tasks/include/clean_chunkserver_with_disk_format.yml:20]: retain pool

File: curve-ansible/server.ini [hunk 5]
```diff
@@ -92,15 +92,14 @@ client_turn_off_health_check=false
 disable_snapshot_clone=true
 chunk_size=16777216
 chunkserver_walfilepool_segment_size=8388608
+retain_chunkfilepool=false
```
Comment (ilixiaocui) [inline curve-ansible/server.ini:95]: retain_pool

File: curve-ansible/roles/clean/files/recycle_chunks.sh [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+dataDir=
+maxChunkId=0
+maxSegmentId=0
+
+function get_max_chunkid_inpool() {
+    ret=`ls -lvr ${dataDir}/chunkserver$1/chunkfilepool`
```
Comment (wu-hanqing) [inline curve-ansible/roles/clean/files/recycle_chunks.sh:24]: ls -lvr 会同时把文件的创建时间，权限都打出来，下面的ret[0]不能直接取出最大的id吧？

File: curve-ansible/roles/clean/files/recycle_chunks.sh [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        maxChunkId=${ret[0]}
+    fi
+}
+
+function get_max_segmentid_inpool() {
+    if [ ! -d ${dataDir}/chunkserver$1/walfilepool ]
+    then
+        return
+    fi
+    ret=`ls -lvr ${dataDir}/chunkserver$1/walfilepool`
```
Comment (wu-hanqing) [inline curve-ansible/roles/clean/files/recycle_chunks.sh:36]: 同上

**Pull request closed without merging**